addons/purchase/static/src/components/monetary_field_no_zero/monetary_field_no_zero.js

```diff
@@ -0,0 +1,28 @@
+/** @odoo-module **/
+
+import {monetaryField, MonetaryField} from "@web/views/fields/monetary/monetary_field";
+import { registry } from "@web/core/registry";
+
+export class MonetaryFieldNoZero extends MonetaryField {
+    static template = "account_accountant.BankRecMonetaryField";
+    static props = {
+        ...MonetaryField.props,
+    };
+
+    /** Override **/
+    get value() {
+        const originalValue = super.value;
+        if (originalValue === 0) {
+            return false;
```

Comment:
IMHO, I prefer `return null` here but that's nitpicking

Reply:
I tried `null`, but got a traceback because something in the formatMonetary expects `false` instead :)